erpnext/controllers/stock_controller.py

```diff
@@ -42,6 +43,18 @@ def make_gl_entries(self, gl_entries=None, repost_future_gle=True, from_repost=F
 			gl_entries = self.get_asset_gl_entry(gl_entries)
 			make_gl_entries(gl_entries, from_repost=from_repost)
 
+	def validate_serialized_batch(self):
+		from erpnext.stock.doctype.serial_no.serial_no import get_serial_nos
+		for d in self.get("items"):
+			if not (d.serial_no and d.batch_no): continue
+
+			serial_nos = get_serial_nos(d.serial_no)
+			for serial_no_data in frappe.get_all("Serial No",
+				filters={"name": ("in", serial_nos)}, fields=["batch_no", "name"]):
+				if serial_no_data.batch_no != d.batch_no:
+					frappe.throw(_("The batch {0} does not belog to serial no {1}")
```

Comment:
The reverse validation should be added, as batch is superset.
"Row #{0}: Serial No {1} does not belong to Batch {2}"